src/main/java/uk/ac/ebi/eva/commons/models/converters/data/DBObjectToSamplesConverter.java

```diff
@@ -0,0 +1,145 @@
+     * 
+     * @param samples The list of samples, if any
+     */
+    public DBObjectToSamplesConverter(List<String> samples) {
+        this(true);
+        setSamples(samples);
+    }
+
+    /**
+     * Create a converter from DBObject to a Map of samples, providing a map
+     * of sample names to the corresponding sample id.
+     *
+     * @param sampleIds Map of samples to sampleId
+     */
+    public DBObjectToSamplesConverter(boolean compressDefaultGenotype, Map<String, Integer> sampleIds) {
+        this(compressDefaultGenotype);
+        setSampleIds(sampleIds);
+    }
+
+
+    @Override
+    public VariantSourceEntry convert(DBObject object) {
+        
+        if (sampleIds == null || sampleIds.isEmpty()) {
+            return new VariantSourceEntry(object.get(FILEID_FIELD).toString(), object.get(STUDYID_FIELD).toString());
+        }
+        
+        BasicDBObject mongoGenotypes = (BasicDBObject) object.get(SAMPLES_FIELD);
+//        int numSamples = idSamples.size();
+        
+        // Temporary file, just to store the samples
+        VariantSourceEntry fileWithSamples = new VariantSourceEntry(object.get(FILEID_FIELD).toString(), 
+                object.get(STUDYID_FIELD).toString());
+
+        // Add the samples to the file
+        for (Map.Entry<String, Integer> entry : sampleIds.entrySet()) {
+            Map<String, String> sampleData = new HashMap<>(1);  //size == 1, only will contain GT field
+            fileWithSamples.addSampleData(entry.getKey(), sampleData);
+        }
+
+        // An array of genotypes is initialized with the most common one
+        String mostCommonGtString = mongoGenotypes.getString("def");
+        if(mostCommonGtString != null) {
+            for (Map.Entry<String, Map<String, String>> entry : fileWithSamples.getSamplesData().entrySet()) {
+                entry.getValue().put("GT", mostCommonGtString);
+            }
+        }
+
+        // Loop through the non-most commmon genotypes, and set their value
+        // in the position specified in the array, such as:
+        // "0|1" : [ 41, 311, 342, 358, 881, 898, 903 ]
+        // genotypes[41], genotypes[311], etc, will be set to "0|1"
+        for (Map.Entry<String, Object> dbo : mongoGenotypes.entrySet()) {
+            if (!dbo.getKey().equals("def")) {
+                String genotype = dbo.getKey().replace("-1", ".");
+                for (Integer sampleId : (List<Integer>) dbo.getValue()) {
+                    if (idSamples.containsKey(sampleId)) {
+                        fileWithSamples.getSamplesData().get(idSamples.get(sampleId)).put("GT", genotype);
+                    }
+                }
+            }
+        }
+
+
+        
+        return fileWithSamples;
+    }
+
+
+    public void setSamples(List<String> samples) {
+        int i = 0;
+        int size = samples == null? 0 : samples.size();
+        sampleIds = new HashMap<>(size);
+        idSamples = new HashMap<>(size);
+        if (samples != null) {
```

Comment:
Just that it can be misused and prone to hidden errors. One thing is that the user passes consciously a empty list of samples, another one is to interpret a null value as an empty list, so it's better to raise an exception because it's not a valid value, that way if there is a misuse or a programming mistake you get a nice big warning.